Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,210 @@
+
+        /// <summary>
+        /// Gets or sets the DeviceList Selection Mode. Defaults to ListViewSelectionMode.Single
+        /// </summary>
+        public ListViewSelectionMode DeviceListSelectionMode
+        {
+            get { return (ListViewSelectionMode)GetValue(DeviceListSelectionModeProperty); }
+            set { SetValue(DeviceListSelectionModeProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="DeviceListSelectionMode"/>.
+        /// </summary>
+        public static readonly DependencyProperty DeviceListSelectionModeProperty = DependencyProperty.Register(
+            nameof(DeviceListSelectionMode),
+            typeof(ListViewSelectionMode),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(ListViewSelectionMode.Single));
+
+        /// <summary>
+        /// Fired when the Remote Device Picker is Closed.
+        /// </summary>
+        public event EventHandler<RemoteDevicePickerEventArgs> RemoteDevicePickerClosed;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteDevicePicker"/> class.
+        /// </summary>
+        public RemoteDevicePicker()
+        {
+            DefaultStyleKey = typeof(RemoteDevicePicker);
+            SecondaryButtonText = "Done";
+            SecondaryButtonClick += RemoteDevicePicker_SecondaryButtonClick;
+            Loading += RemoteDevicePicker_Loading;
+            Loaded += RemoteDevicePicker_Loaded;
+        }
+
+        private void RemoteDevicePicker_SecondaryButtonClick(ContentDialog sender, ContentDialogButtonClickEventArgs args)
+        {
+            ObservableCollection<RemoteSystem> selectedItems = new ObservableCollection<RemoteSystem>();
+            foreach (RemoteSystem sys in _listDevices.SelectedItems)
+            {
+                selectedItems.Add(sys);
+            }
+
+            RemoteDevicePickerEventArgs eventArgs = new RemoteDevicePickerEventArgs(selectedItems);
+            RemoteDevicePickerClosed?.Invoke(this, eventArgs);
+            Hide();
+        }
+
+        /// <inheritdoc />
+        protected override void OnApplyTemplate()
+        {
+            _listDevices = GetTemplateChild("PART_LISTDEVICES") as ListView;
+            _listDeviceTypes = GetTemplateChild("PART_LISTDEVICETYPES") as ComboBox;
+            _progressRing = GetTemplateChild("PART_PROGRESS") as ProgressRing;
+
+            var enumval = Enum.GetValues(typeof(DeviceType)).Cast<DeviceType>();
+            _listDeviceTypes.ItemsSource = enumval.ToList();
+            _listDeviceTypes.SelectedIndex = 0;
+            base.OnApplyTemplate();
+        }
+
+        private void ListDeviceTypes_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            UpdateList();
+        }
+
+        private void RemoteDevicePicker_Loaded(object sender, RoutedEventArgs e)
+        {
+            RemoteSystems = new ObservableCollection<RemoteSystem>();
```

Comment:
Set these two in the constructor or at declaration. No need for this event handler